addons/main/functions/fnc_receiveDamage.sqf

```diff
@@ -85,15 +85,15 @@ if (_newHP isEqualTo 0) exitWith {
     } else {
         // damage to downed units
         private _downedHits = 0;
-        If (_downDamage > 1) then {
+        if (_downDamage > 1) then {
             _downedHits = ((_unit getVariable [QGVAR(downedHits),0]) + 1);
             _unit setVariable [QGVAR(downedHits), _downedHits];
         };
         if (_downDamage == 2 && {_downedHits < GVAR(downedDamageHits)}) exitWith {};
         private _downedHp = _unit getVariable [QGVAR(downedHp), (_maxHp * GVAR(downedDamageHP))];
         private _newDownedHP = (_downedHp - _damage) max 0;
         _unit setVariable [QGVAR(downedHp), _newDownedHP];
-        if (_downedHits >= GVAR(downedDamageHits) || {_newDownedHP isEqualTo 0}) then {
+        if (_downedHits >= GVAR(downedDamageHits) || {_newDownedHP isEqualTo 0 || {_downDamage isEqualTo 0}}) then {
```

Comment:
Straight to death if uncon from another source seems a bit harsh for people trying to avoid that. Might rather add this to the lifestate check on line 70 with `&& {!(_unit getVariable [QGVAR(unconscious), false])}`?

Reply:
Imo there is not much of a good solution for this. We could check if ace_unsconsious is set, as we set it as well for additional ace features. This will work for AI that cannot be uncon just fine.

but for players and ai that can be uncon it will fail, as the other script will most likely set the vanilla uncon state to false thus the unit receiving the "free revive".

This solution is to straight up punish as you saw if that happens. Works well with none uncon ai, and it will be harsh  for players and uncon ai.

we could do additional checks to _not_ kill players here, but that would lead them to be walking zombies again.